bbb/meta-bbb/recipes-core/trebletrouble/files/src/libfft.c

```diff
@@ -0,0 +1,127 @@
+/* libfft.c - fast Fourier transform library
+**
+** Copyright (C) 1989 by Jef Poskanzer.
+**
+** Permission to use, copy, modify, and distribute this software and its
+** documentation for any purpose and without fee is hereby granted, provided
+** that the above copyright notice appear in all copies and that both that
+** copyright notice and this permission notice appear in supporting
+** documentation.  This software is provided "as is" without express or
+** implied warranty.
+*/
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <math.h>
+#include "libfft.h"
+#include "tone.h"
+
+#define MAXFFTSIZE 32768
+#define LOG2_MAXFFTSIZE 15
+
+static int bitreverse[MAXFFTSIZE], bits;
+
+/* initfft - initialize for fast Fourier transform
+**
+** b    power of two such that 2**nu = number of samples
+*/
+int initfft( int b )
+    {
+    register int i, j, k;

```

Comment:
Can we remove it?
> Some developers use Local Register Variables in an attempt to improve gcc's allocation of registers, especially in large functions. In this case the register name is essentially a hint to the register allocator. While in some instances this can generate better code, improvements are subject to the whims of the allocator/optimizers. Since there are no guarantees that your improvements won't be lost, this usage of Local Register Variables is discouraged. 
https://gcc.gnu.org/onlinedocs/gcc/Local-Register-Variables.html